mir-ci/mir_ci/program.py

```diff
@@ -27,23 +33,47 @@ def format_output(name: str, output: str) -> str:
 class ProgramError(RuntimeError):
     pass
 
-class Program:
-    def __init__(self, command: Command, env: Dict[str, str] = {}):
+class Program(Benchmarkable):
+    def __init__(
+            self,
+            command: Command,
+            env: Dict[str, str] = {},
+            systemd_slice: Optional[str] = None):
         if isinstance(command, str):
             self.command: tuple[str, ...] = (command,)
         else:
             self.command = tuple(command)
+
         self.name = self.command[0]
         self.env = env
         self.process: Optional[asyncio.subprocess.Process] = None
         self.process_end: Optional[Awaitable[None]] = None
         self.send_signals_task: Optional[asyncio.Task[None]] = None
         self.output = ''
         self.sigkill_sent = False
+        self.cgroup: Optional[Cgroup] = None
+
+        if systemd_slice is not None:
+            prefix = ("systemd-run", "--user", "--scope", f"--slice={systemd_slice}")
+            self.command = (*prefix, *self.command)
 
     def is_running(self) -> bool:
         return self.process is not None and self.process.returncode is None
 
+    def _get_pid(self) -> int:
+        if self.process:
+            return self.process.pid
+        raise RuntimeError("Attempted to get the PID of the program before it was run")
```

Comment:
Ditto, AttributeError should be enough?

Reply:
Actually, `mypy` gets upset here, saying that process could be `None`

Reply:
It turns out, I don't even need this method to begin with :)